tools/workbench/src/main/java/org/eclipse/rdf4j/workbench/proxy/WorkbenchServlet.java

```diff
@@ -127,7 +134,27 @@ private void handleRequest(final HttpServletRequest req, final HttpServletRespon
 		} catch (UnauthorizedException e) {
 			handleUnauthorizedException(req, resp);
 		} catch (RepositoryConfigException | RepositoryException e) {
-			throw new ServletException(e);
+			if (e.getCause() instanceof RemoteShaclSailValidationException) {
+				Model model = ((RemoteShaclSailValidationException) e.getCause()).validationReportAsModel();
+
+				resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
Perhaps a 409 (conflict) is more specific.

Reply:
409 seems good. Hadn’t heard of that one before.